sentinel/server.js

```diff
@@ -19,15 +20,42 @@ process.on('unhandledRejection', reason => {
   console.error(reason);
 });
 
+const waitForApi = () => new Promise(resolve => {
+  //
+  // This waits forever, with no escape hatch, becayse there is no way currently
+  // to know what API is doing, and migrations could legitimately take days
+  //
+  //
+  const waitLoop = () => {
+    request(`http://localhost:${process.env.API_PORT || 5988}/setup/poll`, (err, response, body) => {
+      if (err) {
+        logger.info('Waiting for API to be ready...');
+        return setTimeout(() => waitLoop(), 10 * 1000);
+      }
+
+      logger.info('Api is ready:', body);
+      resolve();
+    });
+  };
+
+  waitLoop();
+});
+
 serverChecks.check(db.serverUrl)
-  .then(config.init())
+  .then(waitForApi)
   .then(() => {
-    if (!loglevel) {
-      logger.transports.Console.level = config.get('loglevel');
-      logger.debug('loglevel is %s.', logger.transports.Console.level);
-    }
-    require('./src/schedule').checkSchedule();
-    logger.info('startup complete.');
+    // Even requiring this boots translations, so has to be required after
+    // api has booted
+    const config = require('./src/config');
+    return config.init()
+      .then(() => {
+        if (!loglevel) {
+          logger.transports.Console.level = require('./src/config').get('loglevel');
```

Comment:
I think you can use the `config` you required 4 lines above.

Reply:
Whoops sorry, I refactored it so it would be in scope then didn't actually use it 🤣